src/jit/assertionprop.cpp

```diff
@@ -4518,11 +4518,9 @@ ASSERT_TP* Compiler::optComputeAssertionGen()
         GenTree*  jtrue    = nullptr;
 
         // Walk the statement trees in this basic block.
-        for (GenTree* stmt = block->bbTreeList; stmt; stmt = stmt->gtNext)
+        for (GenTreeStmt* stmt = block->firstStmt(); stmt != nullptr; stmt = stmt->getNextStmt())
         {
-            noway_assert(stmt->gtOper == GT_STMT);
-
-            for (GenTree* tree = stmt->gtStmt.gtStmtList; tree; tree = tree->gtNext)
+            for (GenTree* tree = stmt->gtStmtList; tree; tree = tree->gtNext)
```

Comment:
`tree != nullptr` (I wonder if it would be worthwhile adding the necessary machinery to support range for syntax...)

Reply:
> I wonder if it would be worthwhile adding the necessary machinery to support range for syntax...

I thought about it and prototyped it, but decided not to include it here.

Reply:
Fixed in the next commits.

Reply:
Yeah, I tried something like this in the past with flowgraphs blocks. My main concern was (and still is) code that attempts to modify the last while traversing it.

Reply:
I do not think it happens often. 
I know `Compiler::optEarlyProp()` changes `gtNextStmt`, does anybody else do this?

Reply:
No idea. It's just that there's always a risk when traversing a list and hiding the list traversal logic behind range for needs to be done with care - the behavior of the iterator in such cases needs to be clearly specified.